src/Ldap.php

```diff
@@ -1202,10 +1348,13 @@ public function add($dn, array $entry)
             }
         }
 
-        $resource = $this->getResource();
-        ErrorHandler::start(E_WARNING);
-        $isAdded = ldap_add($resource, $dn->toString(), $entry);
-        ErrorHandler::stop();
+        do {
+            $resource = $this->getResource();
+            ErrorHandler::start(E_WARNING);
+            $isAdded = ldap_add($resource, $dn->toString(), $entry);
+            ErrorHandler::stop();
+        } while ($isAdded === false && $this->shouldReconnect($resource));
+
         if ($isAdded === false) {
```

Comment:
This is an example of why we don't need to throw anything in shouldReconnect()

Reply:
Curious, I'm not sure why this ended up being displayed here, it isn't about the change to the tests.